python/lsst/ts/fbs/utils/maintel/make_fieldsurvey_scheduler.py

```diff
@@ -0,0 +1,147 @@
+# This file is part of ts_fbs_utils.
+#
+# Developed for the Vera Rubin Observatory Telescope and Site System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["MakeFieldSurveyScheduler"]
+
+import typing
+
+from rubin_scheduler.scheduler.basis_functions import BaseBasisFunction
+from rubin_scheduler.scheduler.schedulers import CoreScheduler
+from rubin_scheduler.scheduler.surveys import BaseSurvey, FieldSurvey
+
+from ..data import field_survey_centers
+
+
+class MakeFieldSurveyScheduler:
+    """Construct Feature Based Scheduler configuration consisting of an
+    ensemble of field surveys, and optionally, other types of surveys.
+    """
+
+    def __init__(
+        self,
+        nside: int = 32,
+        ntiers: int = 1,
+    ) -> None:
+
+        self.nside = 32
+        self.surveys: typing.List[typing.List[BaseSurvey]] = [
+            [] for _ in range(0, ntiers)
+        ]
+
+    def _load_candidate_fields(self) -> typing.Dict:
+        """Load pointing center data for field surveys."""
+        name_fields, ra_fields, dec_fields = zip(*field_survey_centers.fields)
+        fields = {}
+        for name, ra, dec in zip(name_fields, ra_fields, dec_fields):
+            fields[name] = {"RA": ra, "dec": dec}
+        return fields
+
+    def add_field_surveys(
+        self,
+        tier: int,
+        json_block: str,
+        observation_reason: str,
+        science_program: str,
+        field_names: typing.List[str],
+        basis_functions: typing.List[BaseBasisFunction] = [],
```

Comment:
This is where you will need to pass the survey_name metadata.

Reply:
I would like to understand why T&S cares about setting survey_name from the kwargs ..
In general it's not an issue, but  I am worried that it could be because survey_name (often but not always) sets scheduler_note and that you're still parsing scheduler_note to make science_program and observation_reason and  target_name, instead of these being independent pieces of information. 
And while that is not currently an issue for the surveys now in use, it really needs to be changed by the time we start testing a set of surveys that look like the real operations surveys .. so I'm trying to keep on top of this.
If that's not the reason, I'm happy .. but I guess I would still like to understand the reasoning. From my point of view, the main use of `survey_name` is to give a useful and distinct identifier for the real-time FBS dashboard viewer.

Reply:
Using science_program for the survey_name